src/xunit.netcore.extensions/Discoverers/ConditionalTestDiscoverer.cs

```diff
@@ -0,0 +1,122 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace Xunit.NetCore.Extensions
+{
+    // Internal helper class for code common to conditional test discovery through
+    // [ConditionalFact] and [ConditionalTheory]
+    internal class ConditionalTestDiscoverer
+    {
+        // This helper method evaluates the given condition member names for a given set of test cases.
+        // If any condition member evaluates to 'false', the test cases are marked to be skipped.
+        // The skip reason is the collection of all the condition members that evalated to 'false'.
+        internal static IEnumerable<IXunitTestCase> Discover(
+                                                        ITestFrameworkDiscoveryOptions discoveryOptions,
+                                                        IMessageSink diagnosticMessageSink,
+                                                        ITestMethod testMethod,
+                                                        IEnumerable<IXunitTestCase> testCases,
+                                                        IEnumerable<string> conditionMemberNames)
+        {
+            // A null or empty list of conditionMemberNames is treated as "no conditions".
+            // and the test cases will not be skipped.
+            // Example: [ConditionalFact()] or [ConditionalFact((string[]) null)]
+            int conditionCount = conditionMemberNames == null ? 0 : conditionMemberNames.Count();
+            if (conditionCount == 0)
+            {
+                return testCases;
+            }
+
+            MethodInfo testMethodInfo = testMethod.Method.ToRuntimeMethod();
+            Type testMethodDeclaringType = testMethodInfo.DeclaringType;
+            List<string> falseConditions = new List<string>(conditionCount);
+
+            foreach (string entry in conditionMemberNames)
+            {
+                string conditionMemberName = entry;
+
+                // Null condition member names are silently tolerated
+                if (string.IsNullOrWhiteSpace(conditionMemberName))
+                {
+                    continue;
+                }
+
+                string[] symbols = conditionMemberName.Split('.');
+                Type declaringType = testMethodDeclaringType;
+
+                if (symbols.Length == 2)
```

Comment:
This was a very narrow scenario, that i need for windows nano (with type.member). Ideally this can be a fully qualified name, with namespace.type.member.. or members inside nested types etc.. But I didnt have such use-cases in my need.